### DISCUSSION TITLE
Editare file nella cartella di homeassistant su raspbian

### DISCUSSION THREAD

**Discussion by calcioscacchi** [indomus/forum#1671] [GENERALE - Discussioni generiche]
Ciao a tutti, 

che io sappia, nella versione CORE (installata su un raspberry 3b) non è possibile avere a disposizione l'editor direttamente da piattaforma HA, quindi di solito per l'editing dei file di configurazione utilizzo Putty con comandi da shell (sudo nano). Questo risulta però molto scomodo, come potete immaginare.

Sarebbe molto più interessante modificare (editare, creare file e cartelle) direttamente da raspbian, al quale accedo tramite vnc. Qui però mi da un problema di permessi, perchè non posso integrare il comando sudo. Qualcuno ha qualche suggerimento per aiutarmi? 
Grazie

**Comment by 50494554524F** [comment]
se non ricordo male con visual studio code puoi "collegarti" via ssh, anche con notepad++ c'è il plugin...

**Reply by calcioscacchi** [reply]
provato con notepad++, riesco a collegarmi ma se provo a creare un file, mi da "unable to create file /home/pi/homeassistant/sensors.yaml".

Ho dato le credenziali dello stesso utente che lavora con putty.. ma al solito, senza comando sudo è difficile

**Reply by Scialla** [reply]
Comunque esiste anche `config editor` come integrazione di hacs che ti consente di editare i file direttamente da HA ma, se sei in docker, cambia i permessi ai file che editi in quanto docker agisce da `root`
Da casa uso Visual Studio Code con grande soddisfazione

**Reply by calcioscacchi** [reply]
grazie @Scialla. 

Ho cambiato utente e gruppo della cartella di homeassistant dando come nuovo proprietario l'utente pi:

`sudo chown -R pi:pi homeassistant
`


e poi ho dato permessi di lettura, scrittura e altro a tutti i file contenuti:

`sudo chmod 770 homeassistant`


spero di non aver fatto un casino ^-^, ma sembra funzionare tutto! Grazie!

**Reply by Scialla** [reply]
anche se cambi l'owner, quando modifichi tramite l'integrazione ti ricambia i permessi dei file che modifichi rendendo proprietario root.
Nessun problema, ma se poi usi altro per modificare quei file non riesci più.. devi prima ricambiargli i permessi
Per evitare questo c'è uno script che fa girare docker in venv, ma comporta problemi su alcune parti HW, a me non vedeva più il bluetooth e il collegamento con i GPIO del rasp (la ventola)

**Reply by calcioscacchi** [reply]
mah finchè il problema è legato solo alla modifica di quei file, adesso che so che ci sono permessi a livello di file non credo sarà un problema.

**Reply by xyzmauro** [reply]
Non ho capito se Core ce l’hai installato in docker xchè eventualmente esiste anche il container di VSCode e puoi visualizzarlo poi con un panel_iframe direttamente nella GUI, lo vedi se sei collegato in locale altrimenti 
“If you are accessing Home Assistant over HTTPS using SSL, you cannot access HTTP sites through an iframe panel”

**Reply by calcioscacchi** [reply]
sisi, docker

**Comment by KC-inDomus** [comment]
> Ciao a tutti,
> 
> che io sappia, nella versione CORE (installata su un raspberry 3b) non è possibile avere a disposizione l'editor direttamente da piattaforma HA, quindi di solito per l'editing dei file di configurazione utilizzo Putty con comandi da shell (sudo nano). Questo risulta però molto scomodo, come potete immaginare.
> 
> Sarebbe molto più interessante modificare (editare, creare file e cartelle) direttamente da raspbian, al quale accedo tramite vnc. Qui però mi da un problema di permessi, perchè non posso integrare il comando sudo. Qualcuno ha qualche suggerimento per aiutarmi? Grazie

questa l'hai letta? https://indomus.it/progetti/modificare-i-file-di-configurazione-home-assistant-core-su-raspbian-con-visual-studio-code-vscode/
